### PR TITLE
feat: decouple dataloader

### PR DESCRIPTION
**Pull request by williamstar**

### CODE REVIEW

File: persia/ctx.py [hunk 1]
```diff
@@ -189,27 +214,29 @@ def __init__(self):
...
+        weight_bound (float, optional): embedding value bound, normal will update the embedding locate in [-weight_bound, weight_bound]
+        ctx_status (CtxStatus, optional): represent current context status
+        device_id (int, optional): current cuda device id
+        enable_backward (bool, optional): enable embeddign gradients update
+        backend_worker_size (int, optional): rpc client thread pool size
+        backward_buffer_size (int, optional): backward update buffer size
+        nats_recv_buffer_size (int, optional): nats recv buffer size, a buffer before rectifier
+        rank_id (int, optional): rank id of this process.
+        world_size (int, optional): world size of this cluster.
+        num_forward_workers (int, optional): worker num of sending forward request to servers.
```
Comment (snowpeakz) [inline persia/ctx.py:237]: remove

File: persia/data.py [hunk 2]
```diff
@@ -1,50 +1,158 @@
...
+        # dynamic import the PyForward due to conditional compilation
+        from persia.prelude import PyForward
+
+        self.channel = channel
+        self.timeout = timeout
+        self.num_workers = num_workers
+
+        self.forward_engine = PyForward(
+            forward_buffer_size,
+            not is_training,
```
Comment (snowpeakz) [inline persia/data.py:142]: It seems that the order of args is not correct?
https://github.com/PersiaML/PersiaML-server/pull/15/files#diff-6bbcd9af19f26d0a6c03de8cc2c879e5669376cf17d91c7e79ca6ad0a23a6f14R532

File: persia/data.py [hunk 3]
```diff
@@ -1,50 +1,158 @@
...
+            not shuffle,
```
Comment (snowpeakz) [inline persia/data.py:143]: Is shuffle equal with not in-order? As we are not doing shuffle in our code, our dis-order of data is affected by network transmission and num of data-loader instances. E.g. there is only one  data-loader instance, data will not fully shuffled.

File: persia/data.py (hunk 2 above)
Comment (williamstar) [reply persia/data.py:142]: 🤣 I swap the `shuffle` and `is_training` twice that make the result correctly lol

Comment (williamstar) [reply persia/data.py]: revised at 7360c14036f8f9bb0669b7bf77d45642bb16ff9c

File: persia/ctx.py (hunk 1 above)
Comment (williamstar) [reply persia/ctx.py]: revised at 7360c14

**Review by snowpeakz — APPROVED** [review]

**Pull request merged**